src/invariant/InvariantManagerImpl.cpp

```diff
@@ -208,12 +210,10 @@ InvariantManagerImpl::onInvariantFailure(std::shared_ptr<Invariant> invariant,
                                          std::string const& message,
                                          uint32_t ledger)
 {
-    mMetricsRegistry
-        .NewCounter(
-            {"invariant", "does-not-hold", "count", invariant->getName()})
-        .inc();
-    mFailureInformation[invariant->getName()].lastFailedOnLedger = ledger;
-    mFailureInformation[invariant->getName()].lastFailedWithMessage = message;
+    mInvariantFailureCount.inc();
+    mFailureInformation.emplace(
```

Comment:
This changes the semantics of `mFailureInformation`. This captures the oldest failure rather than the newest since
> Inserts a new element into the container constructed in-place with the given args if there is no element with the key in the container.

according to https://en.cppreference.com/w/cpp/container/map/emplace.

Reply:
I thought we didn't care to have the latest, just an example as we now log, but maybe it's better to keep the latest failure. Let me revert that